tests/expose_test.go

```diff
@@ -468,6 +469,51 @@ var _ = Describe("Expose", func() {
 
 				Eventually(getStatus, 60, 1).Should(Equal(k8sv1.PodSucceeded))
 			})
+
+			It("[test_id:343]Should Verify an exposed service of a VM is not functional after VM deletion.", func() {
+				By("Getting back the cluster IP given for the service")
+				svc, err := virtClient.CoreV1().Services(vm.Namespace).Get(serviceName, k8smetav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				serviceIP := svc.Spec.ClusterIP
+
+				By("Starting a pod which tries to reach the VMI via ClusterIP")
+				job := tests.NewHelloWorldJob(serviceIP, servicePort)
+				job, err = virtClient.CoreV1().Pods(vm.Namespace).Create(job)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Waiting for the pod to report a successful connection attempt")
+				getStatus := func() k8sv1.PodPhase {
+					pod, err := virtClient.CoreV1().Pods(job.Namespace).Get(job.Name, k8smetav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					return pod.Status.Phase
+				}
+
+				Eventually(getStatus, 60, 1).Should(Equal(k8sv1.PodSucceeded))
+
+				By("Deleting the VM.")
+				Expect(virtClient.VirtualMachine(vm.Namespace).Delete(vm.Name, &k8smetav1.DeleteOptions{})).To(Succeed())
+
+				By("Waiting until VM is fully deleted.")
+				Eventually(func() k8smetav1.StatusReason {
+					_, err := virtClient.VirtualMachine(vm.Namespace).Get(vm.Name, &k8smetav1.GetOptions{})
+					Expect(err).To(HaveOccurred())
+					errReason := err.(*errors.StatusError).ErrStatus.Reason
+					return errReason
```

Comment:
you can do `return errors.IsNotFound(err)`. There is a utility function which does that.

Reply:
@SchSeba What say you? Can we skip the endpoints check here, assuming it is done in other tests?
If you do think it is necessary here - no problem, I can add it.

Reply:
Yeah if we check that in other test case it's ok

Reply:
I added the endpoint check anyway.
And replaced the search for the "NotFound" error reason with the utils function.